cinema/views.py

```diff
@@ -34,7 +38,34 @@ class MovieViewSet(viewsets.ModelViewSet):
     queryset = Movie.objects.all()
     serializer_class = MovieSerializer
 
-    def get_serializer_class(self):
+    @staticmethod
+    def _params_to_ints(query_string: str) -> queryset:
+        return [int(str_id) for str_id in query_string.split(",")]
+
+    def get_queryset(self):
+        queryset = self.queryset
```

Comment:
call super method instead of accesing this attribute directly
also solve N+1 problem